src/java/org/apache/cassandra/service/accord/AccordJournal.java

```diff
@@ -1099,7 +1086,8 @@ private void doRun()
                 long waitForEpoch = waitForEpochs.getLong(i);
                 if (!node.topology().hasEpoch(waitForEpoch))
                     break;
-                requestBuffer.addAll(delayedRequests.remove(waitForEpoch));
+                if (null == requests) requests = new ArrayList<>();
```

Comment:
This could at least be allocated to be the size of the delayed requests in the first epoch

Reply:
Done, thanks.